File: SampleAdBlocker Extension/extension/src/pages/background/content-blocker/BlockerData.ts
/**
 * Wrapper class for results
 */
export class BlockerData {
    scripts: string[] = [];

    cssExtended: string[] = [];

    cssInject: string[] = [];

    scriptlets: string[] = [];

    addScript(script?: string) {
        if (script != null && script !== '') {
            this.scripts.push(script);
        }
    }

    addCssExtended(style?: string) {
        if (style != null && style !== '') {
            this.cssExtended.push(style);
        }
    }

    addCssInject(style?: string) {
        if (style != null && style !== '') {
            this.cssInject.push(style);
        }
    }

    addScriptlet(scriptlet?: string) {
        if (scriptlet != null && scriptlet !== '') {
            this.scriptlets.push(scriptlet);
        }
    }
}

File: SampleAdBlocker Extension/BlockerData.swift
import Foundation

// Wrapper result class
class BlockerData: Encodable {
    var scripts = [String]()
    var cssExtended = [String]()
    var cssInject = [String]()
    var scriptlets = [String]()

    func addScript(script: String?) {
        if (script != nil && script != "") {
            scripts.append(script!);
        }
    }

    func addCssExtended(style: String?) {
        if (style != nil && style != "") {
            cssExtended.append(style!);
        }
    }

    func addCssInject(style: String?) {
        if (style != nil && style != "") {
            cssInject.append(style!);
        }
    }

    func addScriptlet(scriptlet: String?) {
        if (scriptlet != nil && scriptlet != "") {
            scriptlets.append(scriptlet!);
        }
    }

    func clear() {
        scripts = [];
        cssExtended = [];
        cssInject = [];
        scriptlets = [];
    }
}
